crates/containerd-shim-wasm/src/sandbox/shim/local.rs

```diff
@@ -418,41 +420,41 @@ impl<T: Instance + Sync + Send, E: EventSender> Task for Local<T, E> {
         #[cfg(feature = "opentelemetry")]
         tracing::Span::current().set_parent(extract_context(&_ctx.metadata));
 
-        Ok(self.task_delete(req)?)
+        Ok(self.task_delete(req).block_on()?)
     }
 
     #[cfg_attr(feature = "tracing", tracing::instrument(skip(self), level = "Info"))]
     fn wait(&self, _ctx: &TtrpcContext, req: WaitRequest) -> TtrpcResult<WaitResponse> {
         debug!("wait: {:?}", req);
 
         #[cfg(feature = "opentelemetry")]
-        {
+        let span_exporter = {
             use tracing::{Level, Span, span};
             let parent_span = Span::current();
             parent_span.set_parent(extract_context(&_ctx.metadata));
 
-            let (tx, rx) = std::sync::mpsc::channel();
-            // Start a thread to export interval span for long wait
-
-            let _ = thread::spawn(move || {
+            async move {
                 loop {
                     let current_span =
                         span!(parent: &parent_span, Level::INFO, "task wait 60s interval");
                     let _enter = current_span.enter();
-                    if rx.recv_timeout(Duration::from_secs(60)).is_ok() {
-                        break;
-                    }
+                    tokio::time::sleep(std::time::Duration::from_secs(60)).await;
                 }
-            });
-            let result = self.task_wait(req)?;
-            tx.send(()).unwrap();
-            Ok(result)
-        }
+            }
+        };
 
         #[cfg(not(feature = "opentelemetry"))]
-        {
-            Ok(self.task_wait(req)?)
+        let span_exporter = std::future::pending::<()>();
+
+        let res = async {
+            tokio::select! {
+                _ = span_exporter => unreachable!(),
```

Comment:
maybe add a comment explaining why `span_exporter` is never expected to end? My understanding is that this async block is solely determined by the completion of `self.task_wait(req)`